core_types/src/chain_header.rs

```diff
@@ -30,6 +30,8 @@ pub struct ChainHeader {
     link: Option<Address>,
     /// Key to the most recent header of the same type, None is valid only for the first of that type
     link_same_type: Option<Address>,
+    /// Key to the header of the previous version of this chain header's entry
+    link_crud: Option<Address>,
```

Comment:
Not sure if `link_crud` is the best name for this as it is only for modification and the existence of this address means that this header is actually a header of a modification event. I think the term `modification` should be more prominent somehow.

Reply:
it also links to the deletion entry, so its not just modification.